src/main/resources/api/paths/calendar/calendars.yaml

```diff
@@ -0,0 +1,131 @@
+get:
+  summary: Search all calendars
+  description: Get all calendars that match the given query
+  operationId: searchCalendars
+  parameters:
+    - in: query
+      name: id
+      required: false
+      style: form
+      explode: true
+      schema:
+        type: array
+        items:
+          type: string
+          format: uuid
+      description:
+        The list of calendar IDs to retrieve, sent as separate parameters (?id=aaaa&id=bbbb...).  If
+        this list is passed, calendars must have an ID in this list in addition to any additional
+        criteria.
+    - in: query
+      name: servicePointId
+      required: false
+      style: form
+      explode: true
+      schema:
+        type: array
+        items:
+          type: string
+          format: uuid
+      description:
+        Filter for calendars that are assigned to a certain service point.  If this parameter is
+        excluded, all service points will be considered/included in the response.  Multiple service
+        points may be specified with form-style query expansions; in this case, calendars that are
+        assigned to any of the provided service points will be returned.
+    - in: query
+      name: startDate
+      required: false
+      schema:
+        type: string
+        format: date
+      description: The first date to consider, inclusively
+    - in: query
+      name: endDate
+      required: false
+      schema:
+        type: string
+        format: date
+      description: The last date to consider, inclusively
+    - in: query
+      name: offset
+      required: false
+      schema:
+        type: integer
+        default: 0
+        minimum: 0
+      description: Skip a certain number of the first values; used for pagination
+    - in: query
+      name: limit
+      required: false
+      schema:
+        type: integer
+        default: 10
+        minimum: 0
+      description:
+        The maximum number of elements returned in the response, used for pagination.  A limit of
+        zero will not include any results (however, totalRecords will still be included) -- to
+        include all results, use a large number such as 2147483647.
```

Comment:
Given that this API already distinguishes itself from the standard FOLIO conventions, I might be tempted to not include a default limit and instead make it a mandatory parameter.

This is because we've had some situations where the default has caused confusion for clients.

It might also be worth considering setting a maximum value that is more practical for the API to accomplish. I think the current maximum was chosen because it was the maximum size of an integer not because it could practically be achieved by implementations.

This is very much a one to ponder piece of feedback as it could be considered controversial to some folks.

Reply:
> Most clients who have been working with FOLIO APIs should probably expect this standard of pagination, and I'd rather this API not be an exception to this.

Ok. I understand not wanting to be an early mover on this topic.

This is one of the challenges of changing conventions, there is no way to do it atomically and so they often don't get changed. The default limit is often inconvenient for clients and can lead to unexpected behaviour.

Reply:
Definitely agree, it's not easy to change conventions.  If we were to change it, I'd personally advocate for defaulting to no limit -- if clients want to limit the response and paginate for their own purposes, that's up to them, but IMO that behavior should be opt-in

Reply:
> I'd personally advocate for defaulting to no limit -- if clients want to limit the response and paginate for their own purposes, that's up to them, but IMO that behavior should be opt-in

Defaulting to no limit (i.e. no paging) is likely ok (maybe better) for small sets of records. 

For large sets of records (especially in APIs that do aggregation) no paging can be challenging to implement and possibly not useful to the client e.g. would any client really want to receive millions of instance records?

Reply:
I agree -- it's definitely a bit of a conundrum, which is why my gut says to leave it as it is.  If clients don't realize there's a limit, they should realize pretty quickly when only ten responses come back

Reply:
> If clients don't realize there's a limit, they should realize pretty quickly when only ten responses come back

Due to how many of the test environments we use have very limited sets of data, especially for reference records, these issue can often take a long time to be reported.